packages/libraries/yoga/src/index.ts

```diff
@@ -206,21 +213,21 @@ export function useHive(clientOrOptions: HiveClient | HivePluginOptions): Plugin
 
             return null;
           },
-          async getPersistedOperation(key, request, context) {
-            const document = await experimental__persistedDocuments.resolve(key);
-            // after we resolve the document we need to update the cache record to contain the resolved document
-            if (document) {
-              const record = contextualCache.get(context);
-              if (record) {
-                record.experimental__documentId = key;
-                record.paramsArgs = {
-                  ...record.paramsArgs,
-                  query: document,
-                };
+          getPersistedOperation(key, _request, context) {
+            return mapMaybePromise(experimental__persistedDocuments.resolve(key), document => {
```

Comment:
I will accept such performance optimization only if you prove me it actually matters.
I know what's the different between Promise and sync return and how it affects performance, but here, to me, it's completely unnecessary and just adds complexity for no reason.

Reply:
![image](https://github.com/user-attachments/assets/a75d847d-9a46-4df4-9086-b949a9764f30)
Any unnecessary `async` affects the performance, that's why for example graphql-js handles async and sync.

Reply:
I think you ignored what I wrote

Reply:
tell me how 0.004 vs 0.015 matters when it comes to pulling persisted operation from a storage

Reply:
Everytime it gets the requests, any promise in the chain causes the entire operation to be deferred to the next tick.
It is not the numbers there but the exponential latency it adds.

Reply:
Like I said, this is the same why graphql-js also handles resolvers in this way. And this is why we have the same logic in the gateway from bottom to the top.

Reply:
Still not convinced

Reply:
in case of graphql-js, it creates 1020120 promises, here, you have 1.

Reply:
Because I want to avoid to add another promise in the gateway for example.
One extra promise cause the entire chain to be promise.

Reply:
I'd love to avoid any extra latency as much as possible as we care about performance from the lowest level to the highest.

Reply:
With a gateway under slight pressure, by waiting for the next tick, you miss the oportunity to start sending the response imediatly. Which means a ready to send response will wait for the other micro-tasks queued before being actually sent.

And that's what can cause very hight latence spike. If a huge sync task is in the queue, the response will seat there for potentially a long time, for no real reason other than the code looking slightly uglier.

Reply:
that's the theory :) What I asked is a proof that this indeed affects the gateway.
We can micro-optimize 100 things and spend 20h on it and see 2ms different on a query that takes 600ms.
What I'm saying is, that I will only allow for optimizations that are actually affecting the gateway and not polluting code, because we think it does.

Reply:
Anyway reverted

Reply:
The other changes are more important than this :)

Reply:
Also one think I need to make clear, the issue is not to improve the performance of getting this from cache. It is the overall performance it affects because it makes the operation async.
In the entire conversation, you are repeating that it is that specific part, no it is not. My goal is not to make here faster, my goal is to avoid promises as much as we could so we won't have unnecessary promises, and get as much as performance from our gateway/yoga etc as we can. It affects the entire workflow because it adds an extra promise to the pipeline. If this is not your concern as Hive Console team, we can implement our own version of this plugin with our concerns.
> but here, to me, it's completely unnecessary
It might be hard for you to read in that case but readability of us is less important than getting more performance.